collector/zfs_linux.go

```diff
@@ -93,14 +97,16 @@ func (c *zfsCollector) parseProcfsFile(reader io.Reader, fmtExt string, handler
 			continue
 		}
 
-		key := fmt.Sprintf("kstat.zfs.misc.%s.%s", fmtExt, parts[0])
-
-		value, err := strconv.ParseUint(parts[2], 10, 64)
-		if err != nil {
-			return fmt.Errorf("could not parse expected integer value for %q", key)
+		// kstat data type (column 2) should be KSTAT_DATA_UINT64 (4), otherwise ignore
+		// TODO: when other KSTAT_DATA_* types arrive, much of this will need to be restructured
+		if parts[1] == "4" {
```

Comment:
I think we should define const for this to make it a bit more readable.

Reply:
good idea! I'll include a comment for the cross-reference to zfsonlinux code, too.